Highbar.Types/Left.cs

```diff
@@ -0,0 +1,191 @@
+using System;
+using System.Collections.Generic;
+
+namespace Highbar.Types
+{
+  public class Left<L, R> : Either<L, R>
+  {
+    private L _value;
+
+    public Left(L value)
+    {
+      _value = value;
+    }
+
+    public override bool IsLeft => true;
+
+    public override bool IsRight => false;
+
+    public override Either<L, R> Alt(Either<L, R> other)
+    {
+      return other;
+    }
+
+    public override Either<L, R> Alt(Func<Either<L, R>> supplier)
+    {
+      Objects.RequireNonNull(supplier, "supplier must not be null");
+
+      return supplier();
+    }
+
+    public override Either<L, R> Alt(Func<L, Either<L, R>> other)
+    {
+      Objects.RequireNonNull(other, "other must not be null");
+
+      return other(default(L));
+    }
+
+    public override Either<L, MR> Ap<MR>(Either<L, Func<R, MR>> other)
+    {
+      return new Left<L, MR>(Either<L, MR>.FromLeft(this));
```

Comment:
After doing some verification on this particular method, I have found the following:

```haskell
instance Applicative (Either e) where
    pure          = Right
    Left  e <*> _ = Left e
    Right f <*> r = fmap f r
```

To translate, the value on the left of the `<*>` is the `other` in this context. So, regardless of what the current instance is, if the `other` is a `Left`, the `other` is returned.